src/app/modules/grammar/grammar-routing.module.ts

```diff
@@ -1,6 +1,7 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 import { GRAMMAR_PATH, QUESTION_GRAMMAR_PATH } from 'src/app/app-routing.constants';
+import { GuardGuard } from 'src/app/core/guard/guard.guard';
```

Comment:
Please, rename it to logical name, e.g. _AuthGuard_

Reply:
Okay, that was the default name. But I'll rename it. Thank you. I would also like to ask, I don't have a guard working on Speaking and Listening routing modules. But I specified them as canActivate: [GuardGuard].